src/systemctl/systemctl.c

```diff
@@ -7495,7 +7500,8 @@ static int halt_now(enum action a) {
         /* The kernel will automaticall flush ATA disks and suchlike
          * on reboot(), but the file systems need to be synce'd
          * explicitly in advance. */
-        (void) sync();
+        if (!strstr(program_invocation_short_name, "halt") && !arg_no_sync)
```

Comment:
Ok, I just wanted to make sure that only halt respects it, as this was explicitely pointed out in Lennarts comment.
I've removed the redundant check now.